### ISSUE TITLE
Can't add a remote from a sql-server started outside of a dolt directory

### ISSUE THREAD

**Issue by timsehn** [dolthub/dolt#3936]
Shell 1:
```
dolt $ dolt sql-server
Starting server with Config HP="localhost:3306"|T="28800000"|R="false"|L="info"
2022-07-26T16:21:08-07:00 INFO [conn 1] NewConnection {DisableClientMultiStatements=false}
2022-07-26T16:21:57-07:00 WARN [conn 1] error running query {connectTime=2022-07-26T16:21:08-07:00, connectionDb=pnp_bug, error=syntax error at position 19 near 'table', query=create table table (pk int, primary key(pk));}
2022-07-26T16:22:24-07:00 WARN [conn 1] error running query {connectTime=2022-07-26T16:21:08-07:00, connectionDb=pnp_bug, error=syntax error at position 20 near 'table', query=call dolt_add(table);}
2022-07-26T16:22:28-07:00 WARN [conn 1] error running query {connectTime=2022-07-26T16:21:08-07:00, connectionDb=pnp_bug, error=column "table1" could not be found in any table in scope, query=call dolt_add(table1);}
2022-07-26T16:27:47-07:00 WARN [conn 1] error running query {connectTime=2022-07-26T16:21:08-07:00, connectionDb=pnp_bug, error=open /Users/timsehn/liquidata/dolt/.dolt/repo_state.json: no such file or directory, query=call dolt_remote('add', 'origin', 'timsehn/pnp_bug');}
```

Shell 2:

```
git $ dolt sql-client
# Welcome to the Dolt MySQL client.
# Statements must be terminated with ';'.
# "exit" or "quit" (or Ctrl-D) to exit.
mysql> create database pnp_bug;
mysql> use pnp_bug;
...
...
...
mysql> call dolt_remote('add', 'origin', 'timsehn/pnp_bug');
Error 1105: open /Users/timsehn/liquidata/dolt/.dolt/repo_state.json: no such file or directory
```

This looks like because it is looking at $CWD for a .dolt direectory but the use is like `cd`ing into the database directory.

**Comment by tbantle22** [comment]
Tried to add a remote from a hosted sql server and got this error:
```
mysql> CALL DOLT_REMOTE('add','origin','https://doltremoteapi.dolthub.com/Dolthub/SHAQ');
ERROR 1105 (HY000): stored procedure "DOLT_REMOTE" does not exist
```
On dolt version 0.40.20

**Comment by fulghum** [comment]
I was able to repro Tim's exact error (`.dolt/repo_state.json: no such file or directory`) by starting a `dolt sql-server` from a directory where I hadn't run `dolt init` yet. I didn't see the issue when I started a `sql-server` from a directory that had been init'ed, but on closer examination, it looks like the remote metadata is being written to the wrong database directory, and `select * from dolt_remotes;` looks like it works, because it's also reading from the wrong directory. 

I haven't been able to repro Taylor's error yet. On a HostedDolt-dev instance, I'm able to execute the `dolt_remote` stored procedure (but it errors out with the same error as Tim found).
